### PR TITLE
Added support for gas consumption measurements

### PR DESCRIPTION
**Pull request by Przemko92**

### CODE REVIEW

**Review by MislavMandaric — COMMENTED** [review]
I left some comments. This looks really good in general, most comments are related to my lack of knowledge on energy sensors in HA and inability to check things myself as my boiler doesn't expose these sensors.

Thank you for this and sorry for waiting so long. 👍

File: custom_components/vaillant_vsmart/sensor.py [hunk 1]
```diff
@@ -79,3 +107,255 @@ def native_unit_of_measurement(self) -> str:
...
+
+    @property
+    def icon(self) -> str | None:
+        return "mdi:heating-coil"
+
+    @property
+    def state_class(self) -> SensorStateClass:
+        """Return state class for this sensor."""
+
+        return SensorStateClass.TOTAL_INCREASING
```
[the remark this answers is not in the document]
Comment (kristoffeys) [reply custom_components/vaillant_vsmart/sensor.py:147]: I checked this and the value of the sensor resets every night at 12PM, so this seems to be the correct state class.

File: custom_components/vaillant_vsmart/sensor.py [hunk 2]
```diff
@@ -79,3 +107,255 @@ def native_unit_of_measurement(self) -> str:
...
+                    self._module.id
+                ).gas_heating.value
+            )
+        }
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return unit of measurement for the gas heating sensor."""
+
+        return ENERGY_WATT_HOUR
```
[the remark this answers is not in the document]
Comment (kristoffeys) [reply custom_components/vaillant_vsmart/sensor.py:172]: I tested this and can confirm that this is not configurable.

**Comment by MislavMandaric** [comment]
Hey @Przemko92. I have some time to spend improving the integration and this PR is something I'd like to merge and release. It's been a while since you made the changes, but if you have some time to look at my comments and resolving conflicts that resulted from my refactor (sorry about that 😅), I'd greatly appreciate it.

If you don't have time right now, would you mind me resolving these conflicts and merging your changes, aligning them with the new naming conventions myself?

**Comment by Przemko92** [comment]
Hey @MislavMandaric 
I stopped using MiGo so I'm not able to test changes anymore. That's why I didn't fix these issues. Maybe somebody would like to get my code and apply the changes.

**Comment by MislavMandaric** [comment]
Thanks for the update!

In that case I'll wrap up this feature, using your code as the base. Also thank you for taking the time to work on this 👍

**Pull request merged**